python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -519,13 +527,27 @@ def _schedule_replica(
                 if scheduling_request.placement_group_strategy
                 else "PACK"
             )
-            pg = self._create_placement_group_fn(
-                scheduling_request.placement_group_bundles,
-                placement_group_strategy,
-                _soft_target_node_id=target_node_id,
-                lifetime="detached",
-                name=scheduling_request.actor_options["name"],
-            )
+            try:
+                pg = self._create_placement_group_fn(
+                    scheduling_request.placement_group_bundles,
+                    placement_group_strategy,
+                    _soft_target_node_id=target_node_id,
+                    lifetime="detached",
+                    name=scheduling_request.actor_options["name"],
+                )
+            except Exception:
+                # We add a defensive exception here, so the controller can
+                # make progress even if the placement group isn't created.
+                # See https://github.com/ray-project/ray/issues/43888.
+                logger.exception(
+                    f"Failed to create a placement group for {replica_id}."
+                )
+                scheduling_request.scheduling_failed = True
+                scheduling_request.scheduling_failed_reason = (
+                    "Failed to create a placement group for replica "
+                    f"{replica_id.unique_id}. See logs for more details."
+                )
+                return
```

Comment:
It also seems to me that `schedule()` should return a list (dictionary) of replicas to stop. Maybe we can reuse this functionality somehow?

Reply:
We synced offline; I'll keep the implementation as-is.